src/Essentials/src/SecureStorage/SecureStorage.android.cs

```diff
@@ -1,395 +1,175 @@
+			var sharedPreferences = EncryptedSharedPreferences.Create(
+				fileName,
+				prefsMainKey,
+				context,
+				EncryptedSharedPreferences.PrefKeyEncryptionScheme.Aes256Siv,
+				EncryptedSharedPreferences.PrefValueEncryptionScheme.Aes256Gcm
+			);
 
-			return keyGenerator.GenerateKey();
+			return sharedPreferences;
 		}
 
-		KeyPair GetAsymmetricKeyPair()
-		{
-			// set that we generated keys on pre-m device.
-			Preferences.Set(useSymmetricPreferenceKey, false, SecureStorage.Alias);
+		static string GetOrCreateMasterKeyApi23OrNewer()
+			=> MasterKeys.GetOrCreate(MasterKeys.Aes256GcmSpec);
 
-			var asymmetricAlias = $"{alias}.asymmetric";
+		static string GetOrCreateMasterKeyApi22OrOlder(Context context)
+		{
+			// older versions don't support symmetric keys. but we still want to always use the KeyStore
 
-			var privateKey = keyStore.GetKey(asymmetricAlias, null)?.JavaCast<IPrivateKey>();
-			var publicKey = keyStore.GetCertificate(asymmetricAlias)?.PublicKey;
+			var asymmetricAlias = $"dotnet.MAUI.asymmetric";
 
-			// Return the existing key if found
-			if (privateKey != null && publicKey != null)
-				return new KeyPair(publicKey, privateKey);
+			// Force to english for known bug in date parsing:
+			// https://issuetracker.google.com/issues/37095309
+			SetLocale(context, Java.Util.Locale.English);
 
-			var originalLocale = Platform.GetLocale();
-			try
-			{
-				// Force to english for known bug in date parsing:
-				// https://issuetracker.google.com/issues/37095309
-				Platform.SetLocale(Java.Util.Locale.English);
+			// Otherwise we create a new key
+			var generator = KeyPairGenerator.GetInstance(
+			  KeyProperties.KeyAlgorithmRsa,
+			  "AndroidKeyStore"); // Const value for secure keystore
 
-				// Otherwise we create a new key
-				var generator = KeyPairGenerator.GetInstance(KeyProperties.KeyAlgorithmRsa, androidKeyStore);
+			// certificates have issue date and expiration date. related to rsa
+			var end = DateTime.UtcNow.AddYears(30);
+			var startDate = new Java.Util.Date();
 
-				var end = DateTime.UtcNow.AddYears(20);
-				var startDate = new Java.Util.Date();
 #pragma warning disable CS0618 // Type or member is obsolete
-				var endDate = new Java.Util.Date(end.Year, end.Month, end.Day);
-#pragma warning restore CS0618 // Type or member is obsolete
+			var endDate = new Java.Util.Date(end.Year, end.Month, end.Day);
 
-#pragma warning disable CS0618
-				var builder = new KeyPairGeneratorSpec.Builder(Platform.AppContext)
-					.SetAlias(asymmetricAlias)
-					.SetSerialNumber(Java.Math.BigInteger.One)
-					.SetSubject(new Javax.Security.Auth.X500.X500Principal($"CN={asymmetricAlias} CA Certificate"))
-					.SetStartDate(startDate)
-					.SetEndDate(endDate);
+			// generates the key pair stuff
+			var builder = new KeyPairGeneratorSpec.Builder(context)
+			  .SetAlias(asymmetricAlias)
+			  .SetSerialNumber(Java.Math.BigInteger.ValueOf(Math.Abs(asymmetricAlias.GetHashCode())))
```

Comment:
does this serial number need to return the same value between app runs? `string.GetHashCode()` is not stable across program executions: https://andrewlock.net/why-is-string-gethashcode-different-each-time-i-run-my-program-in-net-core/